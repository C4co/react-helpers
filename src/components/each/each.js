import React, { Fragment } from "react"
import PropTypes from "prop-types"

export function Each({ items, children }) {
  return (
    <Fragment>
      {items.map((item, index, arr) => {
        return (
          <Fragment key={index}>
            {children(item, index, arr)}
          </Fragment>
        )
      })}
    </Fragment>
  )
}

Each.propTypes = {
  /** Array of items */
  items: PropTypes.array,

  /** children component function(item, index, arr) */
  children: PropTypes.func
}
