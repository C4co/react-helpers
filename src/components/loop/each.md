## Each

Iteration over multiple itens.

#### Usage
```jsx
  import { Each } from "@cn/react-helpers"
```

#### Example:

```jsx
  <Each items={["first", "second", "third"]}>
    {(item, index) => (
      <section>
        <h1> {index} -> {item} </h1>
        <hr/>
      </section>
    )}
  </Each>
```

#### Properties:

| Name | Type | Default |
| --- | --- | --- |
| @items | array | - |
| @children | function(item, index) | - |
