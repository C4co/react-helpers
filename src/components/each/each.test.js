import { mount } from "enzyme"
import { Each } from "./each"
import React from "react"

describe("Each component", () => {
  test("should iterate over each array item.", () => {
    const each = mount(
      <Each items={["first", "second", "third"]}>
        {(item, index) => (
          <h1>
            {index}-{item}
          </h1>
        )}
      </Each>
    )

    expect(each.contains(<h1>0-first</h1>)).toBe(true)
    expect(each.contains(<h1>1-second</h1>)).toBe(true)
    expect(each.contains(<h1>2-third</h1>)).toBe(true)
    expect(each).toMatchSnapshot()
  })
})
