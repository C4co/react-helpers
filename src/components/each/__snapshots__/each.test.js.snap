// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Each component should iterate over each array item. 1`] = `
<Each
  items={
    Array [
      "first",
      "second",
      "third",
    ]
  }
>
  <h1>
    0
    -
    first
  </h1>
  <h1>
    1
    -
    second
  </h1>
  <h1>
    2
    -
    third
  </h1>
</Each>
`;
